### ISSUE TITLE
🐛 The session middleware performs deserialization, which will cause a crash

### ISSUE THREAD

**Issue by lovesky** [gofiber/fiber#1051]
**Fiber version**
v2.2.2

**Issue description**
Save the session to the database, restart the program, and perform session deserialization will cause a crash.

The following is a simple test, there are two functions, the **test1** is **gotiny.Marshal** and then **gotiny.Unmarshal**, and the test2 is direct **gotiny.Unmarshal**. There is no problem executing the two together. But if you comment out the **test1** and run the **test2** directly, an error will be reported: **panic: unknown typ:int**

**Code snippet**

```go
package main

import (
	"fmt"
	"gotiny" // github.com/gofiber/fiber/v2/internal/gotiny
)

type data struct {
	d map[string]interface{}
}

func main() {
	test1()
	test2()
}

func test1() {
	a := data{d: make(map[string]interface{})}
	a.d["user_role"] = 1
	b := gotiny.Marshal(&a)
        // [3 1 9 117 115 101 114 95 114 111 108 101 3 105 110 116 2]
	fmt.Println(b)
	c := data{}
	gotiny.Unmarshal(b, &c)
        // {map[user_role:1]}
	fmt.Println(c)
}

func test2() {
	d := data{}
	e := []byte{3, 1, 9, 117, 115, 101, 114, 95, 114, 111, 108, 101, 3, 105, 110, 116, 2}
        // panic: unknown typ:int
	gotiny.Unmarshal(e, &d)
        // {map[user_role:1]}
	fmt.Println(d)
}
```

**Comment by Fenny** [comment]
I think we need to expose the `Register` fn in session mw to restore / support custom structs <https://github.com/niubaoshu/gotiny/blob/master/register.go#L116>

**Comment by Fenny** [comment]
You can now use the following to solve this issue:
```go
store := session.New()
store.RegisterType(User)
```

Available in `v2.2.4`  https://github.com/gofiber/fiber/releases/tag/v2.2.4
